tests/RobotFramework/tests/cumulocity/registration/registration_lifecycle.robot

```diff
@@ -484,6 +484,18 @@ Entities send to cloud on restart
     ...    minimum=1
     ...    maximum=1
 
+Unexpected message doesn't cause a panic #3134
+    # Registered as service
+    Execute Command    tedge mqtt pub --retain 'te/device/child1//' '{"@type":"service"}'
+    External Identity Should Exist    ${DEVICE_SN}:device:child1    show_info=False
+    Cumulocity.Managed Object Should Have Fragment Values    status\=up
+    Service Health Status Should Be Up    tedge-mapper-c8y
+
+    # Registered as child device
+    Execute Command    tedge mqtt pub --retain 'te/device/child2/service/foo' '{"@type":"child-device"}'
+    Device Should Exist    ${DEVICE_SN}:device:child2:service:foo
```

Comment:
Here I do think the test should fail ... but it doesn't.

Indeed, ` 'te/device/child2/service/foo' '{"@type":"child-device"}'` registers a child device on a service topic.

@albinsuresh Is this expected or a side effect of auto-registration?

Reply:
Though shouldn't auto registration only kick in if you are not registration something...here it is explicitly registering a given type, so there is no need to try to auto detect anything.

Reply:
Indeed, this is what happens: no auto registration kicks in. That's a just bit confusing because applied on a topic matching auto registration requirements.

Reply:
Yes @reubenmiller is right. Auto-registration kicks in only when the source entity for a message isn't explicitly registered up-front. But, since there is an explicit registration happening here, it just doesn't kick in. In fact we advise the users to disable auto-registration if they decide to use explicit registration for any entities.

We don't prevent the users from re-using the default topic scheme while doing explicit registrations either, but yeah it can result in such unexpected/uncommon usages as well. Not sure if we should completely prevent it, as the default topic scheme still provides a starting point/guidance.